src/zoning-district-class/zoning-district-class.repository.ts

```diff
@@ -17,17 +18,44 @@ export class ZoningDistrictClassRepository {
 
   async findMany(): Promise<FindManyRepo> {
     try {
-      return await this.db.query.zoningDistrictClass.findMany();
+      const result = await this.db.query.zoningDistrictClass.findMany({
+        columns: {
+          id: true,
+          description: true,
+          category: true,
+          color: true,
+          url: true,
+        },
+      });
+
+      return result.map((zoningDistrictClass: any) => ({
+        ...zoningDistrictClass,
+        category: zoningDistrictClass.category as ZoningDistrictClassCategory,
+      }));
     } catch {
       throw new DataRetrievalException();
     }
   }
 
   async findById(id: string): Promise<FindByIdRepo | undefined> {
     try {
-      return await this.db.query.zoningDistrictClass.findFirst({
+      const result = await this.db.query.zoningDistrictClass.findFirst({
```

Comment:
Same feedback as above to use select statement.

When converting to a select statement, we can use `limit(1)` to achieve the same effect as `findFirst`. It will return an array of results. We will return either the first item in the array or `undefined` if there are no matching results.

Reply:
> It will return an array of results. We will return either the first item in the array or undefined if there are no matching results.

Because we're changing to an array, we'll get a type error if we don't make some modifications. Here is a link to a code snippet that should resolves these.

https://github.com/NYCPlanning/ae-zoning-api/commit/4d7666117a1d00b1363baec36ea6acdacbe76a57